### PR TITLE
Support comments in arrow functions when fixing

### LINKED ISSUE #251: no-mocha-arrows drops comments in the function head when fixing
All of these comments are dropped when fixing:
```js
it( ( foo /* one */ ) /* two */ => /* three */ { body(); } );
```

This was noticed while porting this rule to eslint-plugin-qunit: https://github.com/platinumazure/eslint-plugin-qunit/pull/88

The fixes there can be easily backported to this plugin.

### PR DESCRIPTION
**Pull request by edg2s**
Fixes #251

### CODE REVIEW

File: lib/rules/no-mocha-arrows.js [hunk 1]
```diff
@@ -16,21 +15,31 @@ module.exports = {
     create(context) {
         const sourceCode = context.getSourceCode();
 
+        // eslint-disable-next-line max-statements
```
Comment (edg2s) [inline lib/rules/no-mocha-arrows.js:18]: Not sure how you feel about this rule, but I didn't see an obvious way to break up this method without passing multiple return values.

File: lib/rules/no-mocha-arrows.js [hunk 2]
```diff
@@ -16,21 +15,31 @@ module.exports = {
...
-            const paramsRightParen = sourceCode.getTokenBefore(sourceCode.getTokenBefore(fn.body));
-            let paramsFullText = sourceCode.text.slice(paramsLeftParen.range[0], paramsRightParen.range[1]);
+            const arrow = sourceCode.getTokenBefore(fn.body);
+            let firstToken = sourceCode.getFirstToken(fn);
+            const beforeArrowToken = sourceCode.getTokenBefore(arrow);
+            let paramsFullText;
+            let params = sourceCode.text.slice(firstToken.range[0], beforeArrowToken.range[1]).trim();
             let functionKeyword = 'function';
+            const beforeArrowComment = sourceCode.text.slice(beforeArrowToken.range[1], arrow.range[0]).trim();
+            const afterArrowComment = sourceCode.text.slice(arrow.range[1], fn.body.range[0]).trim();
```
Comment (lo1tuma) [inline lib/rules/no-mocha-arrows.js:27]: For better readability I would suggest to extract the repeated pattern of `sourceCode.text.slice(start, end).trim()` to a function.

Comment (edg2s) [reply lib/rules/no-mocha-arrows.js:27]: Is
```js
sliceAndTrim( sourceCode.text, beforeArrowToken.range[1], arrow.range[0] );
```
much more concise or legible?

Comment (lo1tuma) [reply lib/rules/no-mocha-arrows.js:27]: I think `sourceCode.text` doesn’t need to be a parameter when we define the new function in the scope of the `create` function. Maybe a better name would be something like `extractSourceTextByRange()`. WDYT?

File: lib/rules/no-mocha-arrows.js (hunk 1 above)
Comment (lo1tuma) [reply lib/rules/no-mocha-arrows.js:18]: It’s not a hard requirement. I don’t have a better approach in mind for now. As long as we have a good test coverage we can always refactor the code later 😉.

**Pull request merged**